### PR TITLE
Restructure documentation

### PR DESCRIPTION
**Pull request by imphil**
Restructure the existing documentation to group the content by intended
audience. This produces four sections:

* An introduction section, relevant to "newcomers" to Ibex.
* An user guide, intended for hardware designers (integrators) and
  software developers who want to integrate Ibex, and develop software
  for it.
* A reference guide, which provides background information on the
  design. This section is essential when working on Ibex, and also
  documents our design decisions.
* A developer guide aimed at people modifying Ibex itself. It consists
  mostly of process and tool documentation: how to run the verification
  after a code change, how to use GitHub, etc.

This commit is large, but text is mostly unchanged. A couple of
introductions and tables of content were added, but no significant
changes to the text have been made. These will be done in follow-ups.

### CODE REVIEW

**Comment by imphil** [comment]
The easiest way to see the effects of this PR is the RTD rendering available at https://ibex-core--1119.org.readthedocs.build/en/1119/.

**Review by tomroberts-lowrisc — APPROVED** [review]
This LGTM. Obviously there are some sections that will need updating (like "Getting started"!) but agree it makes sense to do such updates as follow up PRs.

Might be worth getting one more set of eyes on the content before merging.

File: doc/01_overview/index.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Introduction to Ibex
+====================
+
+Ibex is a production-quality open source 32 bit RISC-V CPU core written in SystemVerilog.
+The CPU core is heavily parametrizable and well suited for embedded control applications.
+Ibex is being extensively verified and has seen multiple tape-outs.
```
Comment (towoe) [inline doc/01_overview/index.rst:6]: Are there any links to the tape-outs in the doc? Quick search for did not show any. Next sentence says more info, so readers might be inclined to search for a list...

**Review by towoe — APPROVED** [review]
I like the restructuring. Makes the documentation more accessible.
I have some nits and comments/questions, but feel free to ignore those :)

File: doc/01_overview/index.rst (hunk 1 above)
Comment (imphil) [reply doc/01_overview/index.rst:6]: No, not yet. I hope to be able to add it as we move along.

File: doc/01_overview/index.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
```
[the remark this answers is not in the document]
Comment (imphil) [reply doc/01_overview/index.rst:7]: Yes, that's intentional.

File: doc/01_overview/targets.rst [hunk 3]
```diff
@@ -0,0 +1,19 @@
+Synthesis Targets
+=================
+
+ASIC Synthesis
+--------------
+
+ASIC synthesis is supported for Ibex.
+The whole design is completely synchronous and uses positive-edge triggered flip-flops, except for the register file, which can be implemented either with latches or with flip-flops.
+See :ref:`register-file` for more details.
+The core occupies an area of roughly 24 kGE when using the latch-based register file and implementing the RV32IMC ISA, or 16 kGE when implementing the RV32EC ISA.
```
[the remark this answers is not in the document]
Comment (imphil) [reply doc/01_overview/targets.rst:10]: It's just moved content. This section will see a significant rewrite shortly anyway, I'll leave it as-is for now. (And occupies sounds OKish to me.)

File: doc/01_overview/compliance.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
+Standards Compliance
+====================
+
+Ibex is a standards-compliant 32b RISC-V processor.
```
[the remark this answers is not in the document]
Comment (imphil) [reply doc/01_overview/compliance.rst:4]: It's moved content, but I changed that.

**Pull request merged**